repos/fdbt-site/src/pages/editFareStageMatching.tsx

```diff
@@ -0,0 +1,404 @@
+import React, { ReactElement, useState } from 'react';
+import BackButton from '../components/BackButton';
+import { renderResetAndAutoPopulateButtons, StopItem } from '../components/MatchingBase';
+import { batchGetStopsByAtcoCode, getServiceByIdAndDataSource } from '../data/auroradb';
+import {
+    removeDuplicateAdjacentStops,
+    sortingWithoutSequenceNumbers,
+    validateSequenceNumbers,
+} from '../utils/apiUtils/matching';
+import CsrfForm from '../components/CsrfForm';
+import ErrorSummary from '../components/ErrorSummary';
+import FormElementWrapper from '../components/FormElementWrapper';
+import {
+    DIRECTION_ATTRIBUTE,
+    EDIT_FARE_STAGE_MATCHING_ATTRIBUTE,
+    MATCHING_JSON_ATTRIBUTE,
+    MATCHING_JSON_META_DATA_ATTRIBUTE,
+} from '../constants/attributes';
+import { ErrorInfo, NextPageContextWithSession } from '../interfaces';
+import { ReturnTicket, SingleTicket, Stop, WithIds } from '../interfaces/matchingJsonTypes';
+import { FullColumnLayout } from '../layout/Layout';
+import { formatStopName, getAndValidateNoc, getCsrfToken, isReturnTicket } from '../utils';
+import { getSessionAttribute } from '../utils/sessions';
+import { isWithErrors } from '../interfaces/typeGuards';
+import WarningSummary from '../components/WarningSummary';
+
+const title = 'Edit fare stages - Create Fares Data Service ';
+const description = 'Edit fare stages page of the Create Fares Data Service';
+const errorId = 'radio-option-single';
+
+interface EditStagesProps {
+    fareStages: string[];
+    errors: ErrorInfo[];
+    csrfToken: string;
+    stops: Stop[];
+    backHref: string;
+    selectedFareStages: {};
+    warning: boolean;
+}
+
+export const getFareStagesFromTicket = (ticket: WithIds<SingleTicket> | WithIds<ReturnTicket>): string[] => {
+    if (isReturnTicket(ticket)) {
+        return ticket.outboundFareZones.map((fareZone) => fareZone.name);
```

Comment:
I think we need to compare the number of inbound fare zones to the number of outbound fare zones, to make sure we get all of them. For example, the outbound could have ABC and the inbound has ABCDE. 

Another scenario - oubound has ABC and inbound has DEF.